File: app/src/main/java/com/example/neon/deardiary/settings/SettingsContract.java
package com.example.neon.deardiary.settings;

import com.example.neon.deardiary.base.BasePresenter;
import com.example.neon.deardiary.base.BaseView;


interface SettingsContract {
    interface View extends BaseView<Presenter> {
        void onDiaryDeleted();
    }

    interface Presenter extends BasePresenter {
        void deleteAll();
    }
}

File: app/src/main/java/com/example/neon/deardiary/settings/SettingsContract.kt
package com.example.neon.deardiary.settings

import com.example.neon.deardiary.base.BaseView


internal interface SettingsContract {
    interface View : BaseView<Presenter> {
        fun onDiaryDeleted()
    }

    interface Presenter {
        fun deleteAll()
    }
}
